include/libtorrent/aux_/rtc_stream.hpp

```diff
@@ -0,0 +1,343 @@
+	void get_option(GettableSocketOption&) {}
+#endif
+
+	template <class GettableSocketOption>
+	void get_option(GettableSocketOption&, error_code&) {}
+
+	void cancel(error_code&)
+	{
+		cancel_handlers(boost::asio::error::operation_aborted);
+	}
+
+	void close();
+	void close(error_code const&) { close(); }
+
+	close_reason_t get_close_reason();
+
+	bool is_open() const;
+
+	void add_read_buffer(void* buf, std::size_t len);
+	void issue_read();
+	void add_write_buffer(void const* buf, std::size_t len);
+	void issue_write();
+	std::size_t read_some(bool clear_buffers);
+
+	endpoint_type local_endpoint() const
+	{
+		error_code ec;
+		return local_endpoint(ec);
+	}
+
+	endpoint_type local_endpoint(error_code& ec) const;
+
+	endpoint_type remote_endpoint() const
+	{
+		error_code ec;
+		return remote_endpoint(ec);
+	}
+
+	endpoint_type remote_endpoint(error_code& ec) const;
+
+	std::size_t available() const;
+	std::size_t available(error_code& /*ec*/) const { return available(); }
+
+	template <class Handler>
+	void async_connect(endpoint_type const& /*endpoint*/, Handler const& handler)
+	{
+		// Dummy
+		handler(error_code{});
+	}
+
+	template <class Mutable_Buffers, class Handler>
+	void async_read_some(Mutable_Buffers const& buffers, Handler const& handler)
+	{
+		if (!is_open())
+		{
+			post(m_io_context, std::bind<void>(handler, boost::asio::error::not_connected, std::size_t(0)));
+			return;
+		}
+
+		TORRENT_ASSERT(!m_read_handler);
+		if (m_read_handler)
+		{
+			post(m_io_context, std::bind<void>(handler, boost::asio::error::operation_not_supported, std::size_t(0)));
+			return;
+		}
+		std::size_t bytes_added = 0;
+		for (auto i = buffer_sequence_begin(buffers)
+			, end(buffer_sequence_end(buffers)); i != end; ++i)
+		{
+			if (i->size() == 0) continue;
+			add_read_buffer(i->data(), i->size());
+			bytes_added += i->size();
+		}
+		if (bytes_added == 0)
+		{
+			// if we're reading 0 bytes, post handler immediately
+			// asio's SSL layer depends on this behavior
+			post(m_io_context, std::bind<void>(handler, error_code(), std::size_t(0)));
+			return;
+		}
+
+		m_read_handler = handler;
+		issue_read();
+	}
+
+	template <class Protocol>
+	void open(Protocol const&, error_code&)
+	{ /* dummy */ }
+
+	template <class Protocol>
+	void open(Protocol const&)
+	{ /* dummy */ }
+
+	template <class Mutable_Buffers>
+	std::size_t read_some(Mutable_Buffers const& buffers, error_code& ec)
+	{
+		TORRENT_ASSERT(!m_read_handler);
+		if (!is_open())
+		{
+			ec = boost::asio::error::not_connected;
+			return 0;
+		}
+
+		if (available() == 0)
+		{
+			ec = boost::asio::error::would_block;
+			return 0;
+		}
+#if TORRENT_USE_ASSERTS
+		size_t buf_size = 0;
+#endif
+
+		for (auto i = buffer_sequence_begin(buffers)
+			, end(buffer_sequence_end(buffers)); i != end; ++i)
+		{
+			add_read_buffer(i->data(), i->size());
+#if TORRENT_USE_ASSERTS
+			buf_size += i->size();
+#endif
+		}
+		std::size_t ret = read_some(true);
```

Comment:
as far as I can tell, `read_some()` is called twice. Once with the argument `false` and once `true`. The only thing this parameter does is to enable or disable two lines of code at the end of the function. It seems it would be a simplification for `read_some()` to not take a `bool` and to just move those two lines here.